### PR TITLE
6 build simpler side pannel

### LINKED ISSUE #6: Add RPA-app and task catalogue
As an RPA Developer, I'd like to add a linked RPA-application and RPA-task to an RPA-Activity that is in the BPMN diagram.

---

### :thought_balloon: Hints & Thoughts
Have a look at the mockup
### :question: Additional Questions to Answer / Consider

---

## :white_check_mark: Conditions of satisfaction

- [ ] Adding an RPA-task and application should only be possible to RPA-Activities
- [ ] When editing an RPA-Activity the RPA-task and application should be saved.
- [ ] A fitting data structure for the tasks from the backend has been set up
- [ ] Should include a visible attractive sidebar

### PR DESCRIPTION
**Pull request by lukashueller**
Fixes #6 

## Changes
**Implements a custom build sidepanel for our BPMN-Modeling-Interface.**
The following functionalities were added:
- select task / gateway or event and change their name in new sidebar
- select the activity and then select the app and the task to be automated using RPA
- all activities and applications are retrieved from MongoDB

## Screenshots
*Screenshot of our new interface with properties sidepanel and our new header navbar.*

image

## Checklist before merge

**Developer's responsibilities**
* [ ] Approved by at least one developer: *enter reviewer here* @SanJSp @WolfgangDaniel @kej-jay 
* [ ] **Pull request build** has passed
* [ ] Production backend has been updated to reflect new features and fix bugs
* [ ] tested locally (in all browsers if frontend)
* [ ] updated the docs
* [ ] added tests where necessary

**Product Owner's responsibilities**
* [ ] No issues have been found on staging environment
* [ ] Product Owner has approved the release

### CODE REVIEW

File: client/src/components/BtnGotoModeler.jsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+const gotoModeler = () => {
+    return (
+        <NavLink to="/modeler">
+            <Button style={{ marginLeft: '30px' }} type="primary">Go to Modeler</Button>
+        </NavLink>
+    );
+}
+
+export default gotoModeler;
```
Comment (SanJSp) [inline client/src/components/BtnGotoModeler.jsx:13]: newline at EOF

File: client/src/components/headerNavbar/HeaderNavbar.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                    </Menu.Item>
+                    <Menu.Item key="3">
+                        Robot File
+                            <Link to="/robotfile" />
+                    </Menu.Item>
+                </Menu>
+            </Header >
+        );
+    }
+}
```
Comment (SanJSp) [inline client/src/components/headerNavbar/HeaderNavbar.jsx:44]: newline at eof

File: client/src/components/headerNavbar/HeaderNavbar.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ * @example 
+ * return <HeaderNavbar selectedKey={2} />
+ */
+
+export default class HeaderNavbar extends Component {
+    render() {
+        return (
+            <Header style={{ background: '#363636' }}>
+                {/* <img className="logo" src={logo_black} style={{ height: '50px' }} alt="Logo" /> */}
+                {console.log(this.props.selectedKey)}
```
Comment (SanJSp) [inline client/src/components/headerNavbar/HeaderNavbar.jsx:22]: remove log

File: client/src/components/properties-panel/propertiesPanelBuilder/PropertiesPanelBuilder.jsx [hunk 4]
```diff
@@ -142,20 +147,21 @@ export default class PropertiesPanelBuilder extends Component {
...
-  updateSelectedTask(event) {
+  updateSelectedTask(value, event) {
     this.setState({
       selectedTask: event.target.value
     }, () => {
       const modeling = this.state['modeler'].get('modeling');
       let { element } = this.state;
       activityDataRetrieval.fetchAndUpdateRPAProperties(this.state['selectedApplication'], event.target.value, modeling, element, this.state['modeler']);
     })
+    // console.log('New Task selected: ' + value + ' for Application: ' + this.state['selectedApplication']);
```
Comment (SanJSp) [inline client/src/components/properties-panel/propertiesPanelBuilder/PropertiesPanelBuilder.jsx:164]: should be removed

File: client/src/components/properties-panel/propertiesPanelBuilder/PropertiesPanelBuilder.jsx [hunk 5]
```diff
@@ -213,54 +219,73 @@ export default class PropertiesPanelBuilder extends Component {
...
+              <br />
+              <PropertiesPanelTaskDropdown
+                listOfTasks={this.state['tasksForSelectedApplication']}
+                onTaskSelection={this.updateSelectedTask}
+                disabled={this.state['disableTaskSelection']} 
+                currentSelection={element.businessObject['$attrs']['arkRPA:task']} />
+            </>
+          )}
         </fieldset>
+
```
Comment (SanJSp) [inline client/src/components/properties-panel/propertiesPanelBuilder/PropertiesPanelBuilder.jsx:273]: remove empty line

File: client/src/pages/Error.js [hunk 6]
```diff
@@ -1,16 +1,19 @@
...
+                <HeaderNavbar selectedKey={0} />
+                <br />
+                <Title style={{ paddingLeft: '30px' }}>Error: Page does not exist!</Title>
+            </Layout>
         </div>
     );
 }
 
-export default Error;
+export default Error;
```
Comment (SanJSp) [inline client/src/pages/Error.js:19]: newline at eof

File: client/src/pages/Home.js [hunk 7]
```diff
@@ -19,4 +20,4 @@ const Home = () => {
   );
 };
 
-export default Home;
+export default Home;
```
Comment (SanJSp) [inline client/src/pages/Home.js:23]: newline at EOF

File: client/src/pages/RobotFile.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+                <br />
+                <Title style={{ paddingLeft: '30px' }}>
+                    In future versions, the Robot Framework code can also be edited directly here.
+                </Title>
+            </Layout>
+        </div>
+    );
+};
+
+export default RobotFile;
```
Comment (SanJSp) [inline client/src/pages/RobotFile.js:22]: newline at eof

**Review by SanJSp — CHANGES_REQUESTED** [review]
After having changed the branch you want to merge into and merging dev into yours, the Pull Request got much smaller. Unfortunately, the selected values of RPATasks and Apps are currently not stored when switching to another activity. Because we have already merge Dev into this branch, this should work and I wish that you fix that (ether Erik or Lukas).

Apart from that I really like the look and I think you did a great job!

**Review by SanJSp — APPROVED** [review]

**Review by kej-jay — APPROVED** [review]

**Pull request merged**